public/app/plugins/datasource/azuremonitor/types/types.ts

```diff
@@ -23,45 +24,6 @@ export interface DatasourceValidationResult {
   title?: string;
 }
 
-/**
- * Azure clouds known to Azure Monitor.
- */
-export enum AzureCloud {
-  Public = 'AzureCloud',
-  China = 'AzureChinaCloud',
-  USGovernment = 'AzureUSGovernment',
-  None = '',
-}
-
-export type AzureAuthType = 'msi' | 'clientsecret' | 'workloadidentity';
-
-export type ConcealedSecret = symbol;
-
-interface AzureCredentialsBase {
-  authType: AzureAuthType;
-}
-
-export interface AzureManagedIdentityCredentials extends AzureCredentialsBase {
-  authType: 'msi';
-}
-
-export interface AzureWorkloadIdentityCredentials extends AzureCredentialsBase {
-  authType: 'workloadidentity';
-}
-
-export interface AzureClientSecretCredentials extends AzureCredentialsBase {
-  authType: 'clientsecret';
-  azureCloud?: string;
-  tenantId?: string;
-  clientId?: string;
-  clientSecret?: string | ConcealedSecret;
-}
-
-export type AzureCredentials =
-  | AzureManagedIdentityCredentials
-  | AzureClientSecretCredentials
-  | AzureWorkloadIdentityCredentials;
-
 export interface AzureDataSourceJsonData extends DataSourceJsonData {
```

Comment:
`azureCredentials` should be added as a property to this type.

Reply:
Once credentials implementation will be moved to the shared library, it won't know about Azure Monitor's `AzureDataSourceJsonData`. The datasource's business logic doesn't need direct access to the credentials property.

We had related conversation in context of ADX: https://github.com/grafana/azure-data-explorer-datasource/issues/530